bin/delete-empty-mirrors.js

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const program = require('commander');
+const Storage = require('storj-service-storage-models');
+const mongoose = require('mongoose');
+const Config = require('../lib/config');
+
+program.version('0.0.1')
+  .option('-c, --config <path_to_config_file>', 'path to the config file')
+  .option('-u, --mongourl <mongo connection string>', 'mongo url connection string')
+  .parse(process.argv);
+
+
+async function deleteEmptyMirrors() {
+  if (!process.env.NODE_ENV) {
+    throw new Error('NODE_ENV is not set');
+  }
+  try {
+    let mongourl;
+    let deleteCount = 0;
+
+    if (program.mongourl) {
+      mongourl = program.mongourl;
+    }
+
+    const config = new Config(process.env.NODE_ENV, program.config);
+    const storage = new Storage(
+      mongourl || config.storage.mongoUrl,
+      config.storage.mongoOpts
+    );
+
+    const Mirror = storage.models.Mirror;
+    const Shard = storage.models.Shard;
+
+    const cursor = Mirror.find()
+      .sort({
+        '_id': 1
+      })
+      .cursor();
+
+    const chunkOfMirrors = [];
+    const chunkSize = 5;
+
+    cursor.on('data', mirror => {
+      chunkOfMirrors.push(mirror);
+      if (chunkOfMirrors.length === chunkSize) {
+        cursor.pause();
+      }
+    });
+
+    cursor.on('pause', async () =>{
+      for (const mirror of chunkOfMirrors) {
+        try {
+          const { shardHash } = mirror;
+          const shard = await Shard.findOne({ hash: shardHash });
+          if (!shard) {
+            await mirror.remove();
+            console.log('deleted mirror: ', mirror._id);
+            deleteCount += 1;
+          }
+        } catch (err) {
+          console.error('Error processing mirror: ', mirror._id);
+          console.error('Error: ', err.message);
+          cursor.close();
+        }
+      }
+      cursor.resume();
+    });
+
```

Comment:
`.once('error')` listener is missing, it is preferable to add it and `emit` an error if in 'pause' event something happens.

Thus, you can:
- centralize `cursor.close()` and error handling in just one place. 
- avoid mixing `async-await` with callbacks, which often adds more cognitive complexity.

Also, if an internal error happens inside the cursor you should handle it or it will be thrown as an unhandled error, added to the fact that we will not print the last deleted mirror, losing the track of where we were the last time and the point where we should start next time.

Reply:
Added